Foundation/NSOrderedSet.swift

```diff
@@ -9,23 +9,26 @@
 
 /****************       Immutable Ordered Set   ****************/
 open class NSOrderedSet : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, ExpressibleByArrayLiteral {
-    internal var _storage: Set<NSObject>
-    internal var _orderedStorage: [NSObject]
+    internal var _storage: NSMutableSet
+    internal var _orderedStorage: NSMutableArray
     
     open override func copy() -> Any {
         return copy(with: nil)
     }
     
     open func copy(with zone: NSZone? = nil) -> Any {
-        NSUnimplemented()
+        // Return self for immutable type
+        return self
     }
 
     open override func mutableCopy() -> Any {
         return mutableCopy(with: nil)
     }
 
     open func mutableCopy(with zone: NSZone? = nil) -> Any {
-        NSUnimplemented()
+        let mutableSelf = NSMutableOrderedSet(capacity: count)
+        mutableSelf.addObjects(from: self.array)
+        return mutableSelf
```

Comment:
this is sufficient, but doesn't take advantage of the fact that we already know the elements are a set, so everything re-hashes, etc.

Reply:
I looked at how `NSArray` is implemented and it copies over the `_storage` to the next object - should I use a similar mechanism here?

Reply:
arrays don't get to make extra assumptions about their contents -- my point here was that the its possible to avoid the set-membership check (the set will still have to hash etc to maintain correct layout) but since we're already a runtime set, we already have satisfied uniqueness.  its probably a small win -- but for common currency types, small wins matter a bit more.